projects/map/src/lib/map/map.component.ts

```diff
@@ -15,15 +18,51 @@ import { MapService } from '../map-service/map.service';
     }
   `],
 })
-export class MapComponent implements AfterViewInit {
+export class MapComponent implements AfterViewInit, OnDestroy {
+  private static BOOKMARK_DESCRIPTOR: BookmarkFragmentStringDescriptor = { type: 'string', identifier: '' };
+
+  private destroyed = new Subject();
 
   constructor(
     private el: ElementRef,
     private mapService: MapService,
-  ) { }
+    private bookmarkService: BookmarkService,
+  ) {
+    combineLatest([ this.mapService.getMapViewDetails$(), this.mapService.getUnitsOfMeasure$() ])
+      .pipe(takeUntil(this.destroyed))
+      .subscribe(([ info, measure ]) => {
+        if (info.center === undefined || info.center === null) {
+          return;
+        }
+
+        const precision = MapSizeHelper.getCoordinatePrecision(measure);
+        this.bookmarkService.updateFragment(MapComponent.BOOKMARK_DESCRIPTOR, `${info.center[0].toFixed(precision)},${info.center[1].toFixed(precision)},${info.zoomLevel.toFixed(1)}`);
+      });
+
+    this.bookmarkService.registerFragment(MapComponent.BOOKMARK_DESCRIPTOR)
+      .pipe(takeUntil(this.destroyed))
+      .subscribe((descriptor) => {
+        if (descriptor === undefined) {
```

Comment:
Isn't there a loop here, when bookmark changes, the map is centered, but on changing the map the bookmark is updated again?

Reply:
No, the bookmark code will not notify the bookmark has changed, as it hasn't changed noticably.

Reply:
There is still some "loop" here, not infinite but it goes back and forth 1 or 2 times. You notice this when zooming in a little bit, the map starts to zoom but bounces back en zooms in further because of rounding.